src/components/Modal/ModalHeader/ModalHeader.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import clsx from 'clsx';
+import type {ModalHeaderProps} from './ModalHeader.types';
+import './ModalHeader.scss';
+import {Typography} from '~/components';
+
+const ModalHeaderForwardRef: React.ForwardRefRenderFunction<HTMLDivElement, ModalHeaderProps> = ({
+    title,
+    children,
+    ...props
+}, ref) => {
+    return (
+        <div
+            ref={ref}
+            className={clsx('moonstone-modal-header')}
+            {...props}
+        >
+            <Typography variant="heading" weight="bold" aria-label={title} role="heading" aria-level={4}>
```

Comment:
- I guess you don't need to use `aria-label` as the typography component already wrap the text
- If you want to render a `h4` you can use `component="h4"` instead of `area-level=4`
- Don't render `Typography` if no `title` is provided